modules/apps/sharing/sharing-notifications/src/main/java/com/liferay/sharing/notifications/internal/service/NotificationsSharingEntryLocalServiceWrapper.java

```diff
@@ -0,0 +1,193 @@
+import com.liferay.portal.kernel.util.PrefsPropsUtil;
+import com.liferay.portal.kernel.util.PropsKeys;
+import com.liferay.portal.kernel.util.ResourceBundleLoader;
+import com.liferay.portal.kernel.util.ResourceBundleUtil;
+import com.liferay.portal.kernel.util.SubscriptionSender;
+import com.liferay.sharing.constants.SharingEntryActionKey;
+import com.liferay.sharing.constants.SharingPortletKeys;
+import com.liferay.sharing.model.SharingEntry;
+import com.liferay.sharing.notifications.internal.notifications.SharingNotificationMessageProvider;
+import com.liferay.sharing.notifications.internal.util.NotificationsSharingEntryUtil;
+import com.liferay.sharing.service.SharingEntryLocalService;
+import com.liferay.sharing.service.SharingEntryLocalServiceWrapper;
+
+import java.util.Collection;
+import java.util.Date;
+import java.util.ResourceBundle;
+
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Reference;
+
+/**
+ * @author Alejandro Tardín
+ */
+@Component(immediate = true, service = ServiceWrapper.class)
+public class NotificationsSharingEntryLocalServiceWrapper
+	extends SharingEntryLocalServiceWrapper {
+
+	public NotificationsSharingEntryLocalServiceWrapper() {
+		super(null);
+	}
+
+	public NotificationsSharingEntryLocalServiceWrapper(
+		SharingEntryLocalService sharingEntryLocalService) {
+
+		super(sharingEntryLocalService);
+	}
+
+	@Override
+	public SharingEntry addSharingEntry(
+			long fromUserId, long toUserId, long classNameId, long classPK,
+			long groupId, boolean shareable,
+			Collection<SharingEntryActionKey> sharingEntryActionKeys,
+			Date expirationDate, ServiceContext serviceContext)
+		throws PortalException {
+
+		SharingEntry sharingEntry = super.addSharingEntry(
+			fromUserId, toUserId, classNameId, classPK, groupId, shareable,
+			sharingEntryActionKeys, expirationDate, serviceContext);
+
+		_sendNotificationEvent(
+			sharingEntry,
+			UserNotificationDefinition.NOTIFICATION_TYPE_ADD_ENTRY,
+			serviceContext);
+
+		return sharingEntry;
+	}
+
+	@Override
+	public SharingEntry updateSharingEntry(
+			long sharingEntryId,
+			Collection<SharingEntryActionKey> sharingEntryActionKeys,
+			ServiceContext serviceContext)
+		throws PortalException {
+
+		SharingEntry sharingEntry = super.updateSharingEntry(
+			sharingEntryId, sharingEntryActionKeys, serviceContext);
+
+		_sendNotificationEvent(
+			sharingEntry,
+			UserNotificationDefinition.NOTIFICATION_TYPE_UPDATE_ENTRY,
+			serviceContext);
+
+		return sharingEntry;
+	}
+
+	private String _getMessageBody(
+			SharingEntry sharingEntry, User user, String entryURL)
+		throws PortalException {
+
+		ResourceBundle resourceBundle =
+			_resourceBundleLoader.loadResourceBundle(user.getLocale());
+
+		AssetRenderer<?> assetRenderer =
+			NotificationsSharingEntryUtil.getAssetRenderer(sharingEntry);
+
+		String linkText = ResourceBundleUtil.getString(
```

Comment:
I think `assetRenderer` could be null and it can throw a NPE when invoking `assetRenderer.getTitle`

Reply:
I think it would be easier if we add here a `@Reference SharingNotificationMessageProvider` and invoke the method `getSharingEntryAssetTitle` making it public.